### PR TITLE
Fixes #49 backend to manage pending transactions

### LINKED ISSUE #49: Backend to manage pending transactions
**Feature description:**
Need a backend to manage the pending transactions. No image available (for now, check the electron app).

**Possible implementation:**

### PR DESCRIPTION
**Pull request by e1Ru1o**

### CODE REVIEW

**Comment by olemis** [comment]
@stdevRulo conflicts with `develop` .

File: src/coin/skycoin/models/account.go [hunk 1]
```diff
@@ -63,8 +63,20 @@ func (wlt RemoteWallet) ScanUnspentOutputs() core.TransactionOutputIterator { //
 	return nil
 }
 
-func (wlt RemoteWallet) ListTransactions() core.TransactionIterator { //------TODO
-	return nil
+func (wlt RemoteWallet) ListTransactions() core.TransactionIterator {
```
Comment (olemis) [inline src/coin/skycoin/models/account.go]: Should be factored out into a new method defined in `CDryptoAccount` interface o be implemented in `RemoteWallet` .

File: src/coin/skycoin/models/coin.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	Timestamp core.TransactionTimestamp
+	Status    core.TransactionStatus
+	Inputs    []core.TransactionInput
+	Outputs   []core.TransactionOutput
+	Id        string
+	Fee       uint64
+} 
+ 
+func (txn *SkycoinTransaction) SupportedAssets() []string { 
+  	return []string{"SKY", "SKYCH"}
```
Comment (olemis) [inline src/coin/skycoin/models/coin.go]: use variaables in `params`

File: src/coin/skycoin/models/coin.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+ 
+import ( 
+  	"github.com/fibercrypto/FiberCryptoWallet/src/core" 
+) 
+ 
+/* 
+SkycoinTransaction 
+*/ 
+type SkycoinTransaction struct{ //Implements Transaction interface
+	Timestamp core.TransactionTimestamp
```
Comment (olemis) [inline src/coin/skycoin/models/coin.go]: No fields . Just wrap Skycoin core (or readable) types so as to satisfy interface methods.

**Review by olemis — CHANGES_REQUESTED** [review]

**Comment by olemis** [comment]
@stdevRulo Travis failures . Pls fix before merge

File: src/coin/skycoin/models/account.go [hunk 4]
```diff
@@ -66,13 +66,26 @@ func (wlt RemoteWallet) ScanUnspentOutputs() core.TransactionOutputIterator { //
...
 
-func (wlt RemoteWallet) ListTransactions() core.TransactionIterator { //------TODO
+func (wlt RemoteWallet) ListTransactions() core.TransactionIterator { 
 	return nil
 }
 func (wlt RemoteWallet) ListPendingTransactions() core.TransactionIterator { //------TODO
 	return nil
 }
 
+func (wlt RemoteWallet) ListPendingTransactions() core.TransactionIterator {
```
Comment (olemis) [inline src/coin/skycoin/models/account.go]: This function should return an error on failure

**Review by olemis — CHANGES_REQUESTED** [review]

File: src/coin/skycoin/models/coin.go [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+	//TODO:
+	return false 
+} 
+
+func (sto *SkycoinTransactionOutput) GetAddress() core.Address { 
+	return SkycoinAddress{address: sto.Output.Address}
+} 
+
+func (sto *SkycoinTransactionOutput) GetCoins(ticker string) uint64 { 
+	//TODO: return err here
```
Comment (olemis) [inline src/coin/skycoin/models/coin.go]: Return an error instance on failure

**Review by olemis — CHANGES_REQUESTED** [review]

**Comment by olemis** [comment]
@stdevMac can you take a look into this ? Travis failing

**Comment by olemis** [comment]
@stdevRulo conflicts

**Review by olemis — CHANGES_REQUESTED** [review]
Use parameterized `AltcoinQuotient()` everywhere needed e.g.

```sh
$ grep -nr 1000000 src/
src//util/util.go:14:	return uint64(coin * 1000000), err
src//models/blockchainModels.go:42:	bs.infoRequester = *skycoin.NewSkycoinBlockchainStatus(1000000) //FIXME: set correct value
src//models/history/historyManager.go:177:				val = val / 1000000
src//models/history/historyManager.go:205:				val = val / 1000000
src//models/history/historyManager.go:213:				val = val / 1000000
```

**Comment by e1Ru1o** [comment]
> Use parameterized `AltcoinQuotient()` everywhere needed e.g.
> 
> ```shell
> $ grep -nr 1000000 src/
> src//util/util.go:14:	return uint64(coin * 1000000), err
> src//models/blockchainModels.go:42:	bs.infoRequester = *skycoin.NewSkycoinBlockchainStatus(1000000) //FIXME: set correct value
> src//models/history/historyManager.go:177:				val = val / 1000000
> src//models/history/historyManager.go:205:				val = val / 1000000
> src//models/history/historyManager.go:213:				val = val / 1000000
> ```

all fixed except
>```shell
> src//models/blockchainModels.go:42:	bs.infoRequester = *skycoin.NewSkycoinBlockchainStatus(1000000) //FIXME: set correct value
>```
Because it is not related with any coin value calculations

**Review by olemis — APPROVED** [review]

**Pull request merged**